app/models/sponsor.rb

```diff
@@ -11,6 +11,7 @@ class Sponsor < ApplicationRecord
 
   scope :published, -> { where(published: true) }
   scope :with_footer_image, -> { joins(:footer_logo_attachment) }
+  scope :with_banner_ad, -> { joins(:banner_ad_attachment) }
```

Comment:
@codyjamesbrooks I am wondering if you know about the `with_attached_` helper methods that come with ActiveStorage: https://stackoverflow.com/questions/55329098/include-active-storage-attachments-in-active-record-query.